lib/jxl/encode.cc

```diff
@@ -418,6 +427,24 @@ void JxlEncoderInitBasicInfo(JxlBasicInfo* info) {
   info->animation.have_timecodes = JXL_FALSE;
 }
 
+void JxlEncoderInitFrameHeader(JxlFrameHeader* info) {
+  info->duration = 0;
+  info->timecode = 0;
+  info->name_length = 0;
+  info->is_last = JXL_FALSE;
+  info->layer_info.crop_x0 = 0;
+  info->layer_info.crop_y0 = 0;
+  // TODO(lode): must these be set to image size instead, or will caller do
+  // this?
+  info->layer_info.xsize = 0;
+  info->layer_info.ysize = 0;
+  info->layer_info.blend_info.blendmode = JXL_BLEND_REPLACE;
```

Comment:
I would pick a combination of default values that makes sense in at least some use cases. Either a duration of one tick and kReplace blending (so fixed fps animation is the default) or a duration of zero and kBlend blending (so full-sized layers of a composite still is the default). I have no strong preference on which of these is the default, but the combination of zero duration and kReplace does not make sense so I wouldn't use that as the default.

Reply:
I've thought about the options here, and was going to add JxlBasicInfo as input to copy the crop xsize from, but after trying out the different options, have chosen a different alternative instead.

Option 1: set up the default to be sensible for non-animation, with JXL_BLEND_BLEND: this is not a good option, because adding multiple frames for animation is more common, and with the blend default, the user has to manually undo a lot of fields to enable animation.

Option 2: set up the default to be sensible for animation. This is slightly more useful than option 1, but has the issue of giving an invalid state if the basic info does not have "have_animation" in it. And making the initialization depend on the basic info for this only makes the function more complex to use for the user (as they may still want a composite still frame in this case too). So this is also not a good option

Option 3: set the default to the default values that are in the specification. the JXL specification happens to have the combination of duration 0, and JXL_BLEND_REPLACE, as its default. This option is not so bad imho: the values are mostly set up for animation, the user just has to set duration (and possibly timecode), that's not too bad at all. When the user wants a composite still, they need some more knowledge of all the other fields, but at least they don't have to touch animation related fields (setting duration back to 0) for this.

So I went with option 3: all defaults for the spec. As for crop, xsize and ysize default to 0 in the spec, so setting them to this. To use cropping, we could add the have_crop field like the spec has (and then the user sets the desired crop xsize and ysize when the set this field to true), but that's unrelated to this MR

Reply:
We could define have_crop to be implicitly equal to `layer_info.xsize && layer_info.ysize`, i.e. if you don't touch them then have_crop is false and the frame is implicitly canvas-sized.